fedimint-core/src/db/mod.rs

```diff
@@ -191,13 +191,23 @@ impl Database {
     pub async fn autocommit<'s: 'dt, 'dt, F, T, E>(
         &'s self,
         tx_fn: F,
-        max_retries: Option<usize>,
+        max_attempts: Option<usize>,
     ) -> Result<T, AutocommitError<E>>
     where
         for<'a> F: Fn(&'a mut DatabaseTransaction<'dt>) -> BoxFuture<'a, Result<T, E>>,
     {
-        let mut retries: usize = 0;
+        assert_ne!(max_attempts, Some(0));
+        let mut curr_attempts: usize = 0;
+
         loop {
+            // The `checked_add()` function is used to catch the `usize` overflow.
+            // With `usize=32bit` and an assumed time of 1ms per iteration, this would crash
+            // after ~50 days. But if that's the case, something else must be wrong.
+            // With `usize=64bit` it would take much longer, obviously.
```

Comment:
Love this attention to detail!

Reply:
thanks :) I think at some point we talked about this a while back